sentry-apollo/src/main/java/io/sentry/apollo/SentryApolloInterceptor.kt

```diff
@@ -0,0 +1,105 @@
+package io.sentry.apollo
+
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.Subscription
+import com.apollographql.apollo.exception.ApolloException
+import com.apollographql.apollo.interceptor.ApolloInterceptor
+import com.apollographql.apollo.interceptor.ApolloInterceptor.CallBack
+import com.apollographql.apollo.interceptor.ApolloInterceptor.FetchSourceType
+import com.apollographql.apollo.interceptor.ApolloInterceptor.InterceptorRequest
+import com.apollographql.apollo.interceptor.ApolloInterceptor.InterceptorResponse
+import com.apollographql.apollo.interceptor.ApolloInterceptorChain
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import java.util.concurrent.Executor
+
+class SentryApolloInterceptor(
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val beforeSpan: BeforeSpanCallback? = null
+) : ApolloInterceptor {
+
+    constructor(hub: IHub) : this(hub, null)
+    constructor(beforeSpan: BeforeSpanCallback) : this(HubAdapter.getInstance(), beforeSpan)
+
+    override fun interceptAsync(request: InterceptorRequest, chain: ApolloInterceptorChain, dispatcher: Executor, callBack: CallBack) {
+        val activeSpan = hub.span
+        if (activeSpan == null) {
+            chain.proceedAsync(request, dispatcher, callBack)
+        } else {
+            val span = startChild(request, activeSpan)
+            val sentryTraceHeader = span.toSentryTrace()
+
+            // we have no access to URI, no way to verify tracing origins
+            val headers = request.requestHeaders.toBuilder().addHeader(sentryTraceHeader.name, sentryTraceHeader.value).build()
+            val requestWithHeader = request.toBuilder().requestHeaders(headers).build()
+            span.setTag("operationId", requestWithHeader.operation.operationId())
+            span.setTag("variables", requestWithHeader.operation.variables().valueMap().toString())
+
+            chain.proceedAsync(requestWithHeader, dispatcher, object : CallBack {
+                override fun onResponse(response: InterceptorResponse) {
+                    span.status = response.httpResponse.map { SpanStatus.fromHttpStatusCode(it.code(), SpanStatus.INTERNAL_ERROR) }
+                        .or(SpanStatus.UNKNOWN)
```

Comment:
Sounds like we use `INTERNAL_ERROR` as the default/fallback on the map so it won't even get to `or` here? Is this to avoid nulls?

I think we should also use `UNKNOWN` on the `fromHttpStatusCode`'s default arg:

```suggestion
                    span.status = response.httpResponse.map { SpanStatus.fromHttpStatusCode(it.code(), SpanStatus.UNKNOWN) }
                        .or(SpanStatus.UNKNOWN)
```

Reply:
`INTERNAL_ERROR` if it throws, or do not set anything if `it.code()` does not match, discussed here https://github.com/getsentry/develop/issues/307#issuecomment-822240763